Code/GraphMol/MarvinParse/MarvinParser.h

```diff
@@ -0,0 +1,81 @@
+//
+//  Copyright (C) 2002-2022 Greg Landrum and other RDKit contributors
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#ifndef RD_MARVINPARSER_H
+#define RD_MARVINPARSER_H
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/ChemReactions/Reaction.h>
+
+#include <string>
+#include <iostream>
+
+namespace RDKit {
+
+RDKIT_FILEPARSERS_EXPORT void *MrvFileParser(const std::string &fname,
```

Comment:
We should probably never use void * in C++ anymore.

Our CDXML reader has the same issue, we decided to break the readers into two functions

```
RDKIT_FILEPARSERS_EXPORT std::vector<std::unique_ptr<RWMol>> CDXMLToMols(
    const std::string &cdxml, bool sanitize = true, bool removeHs = true);
```

Which can get the mols out of a cdxml file whether they are reactions or not.  Each molecule is tagged with the scheme info if it is a reaction.

And in ChemReactions

```
RDKIT_CHEMREACTIONS_EXPORT std::vector<std::unique_ptr<ChemicalReaction>>
CDXMLToChemicalReactions(const std::string &rxnBlock, bool sanitize = false,
                         bool removeHs = false);
```


We should have had a utility function IsCDXMLReaction(mols) and CDXMLMolsToReation(mols) to complete the analysis, but Greg thought that you should now whether the file contained reactions or not.

In any case, void * is not a good choice, I would choose std::variant in this case. This  would also remove the isReaction reference as well I think.

Reply:
@greglandrum thoughts?

Reply:
You have some of this functionality below, perhaps we could use the same API as the CDXML parsers?

Reply:
Love the Marvin Sketch reader btw, you might convince me to finish the CDX binary reader at some point.

Reply:
I do like being able to be explicit about what you expect from the file and generating errors if the file contains something else.

If we wanted to (also?) have something which can return either reaction(s) or mol(s), then maybe a function returning a `std::variant` is the best approach.

Reply:
and, since I wasn't explicit: returning a `void *` is bad.

Reply:
The reason the MRB parser/generator is NOT inside FileParsers is because MRB files can be either Mols or Reactions. (like smiles).  If it is inside FileParsers , then FileParsers (the library) would have to access ChemicalReactions, which already references FileParsers.  This creates a cyclic reference error/issue. So I put it at the same level as Smiles